Docusaurus: README update (#33849)
airbytehq/airbyte: docusaurus/README.md

# Documentation and Docusaurus

> We use [Docusaurus](https://docusaurus.io) to build Airbyte

Reviewer: ```suggestion
We use [Docusaurus](https://docusaurus.io) to build Airbyte's
```

yarn build
```

`yarn build` will build Docusaurus site in `docusaurus/build` directory.

## Developing Locally

If you want to make changes to the documentation, you can run docusaurus locally in a way that would
listen to any source docs changes and live-reload them:

```bash
yarn start # any changes will automatically be reflected in your browser!
```

All the content for docs.airbyte.com lives in the `/docs` directory in this repo. All files are

> markdown. Make changes or add new files, and you should see them in your browser!

Reviewer: Caveat: adding new files and seeing them requires changing the sidebar (thanks for separating out the next section to make it clearer!), so you can't see this with yarn start. I believe if you need to change the sidebar, you have to run `build` again (unconfirmed)

Author: Mmmm, I'll double check!

## Changing Navigation Structure

If you have created any new files, be sure to add them manually to the table of contents found here
in [`sidebars.js`](https://github.com/airbytehq/airbyte/blob/master/docusaurus/sidebars.js)

## Contributing

We welcome documentation updates! If you'd like to contribute a change, please make sure to:

- Run `yarn build` and check that all build steps are successful.
- Push your changes into a pull request, and follow the PR template instructions.

When you make a pull request, Vercel will automatically build a test instance of the full docs site
and link it in the pull request for review.

### Checking for broken links

> Airbyte docs site checks link with Docusaurus at build time, and with an additional GitHub action

Reviewer: ```suggestion
Airbyte's docs site checks links with Docusaurus at build time, and with an additional GitHub action
```

periodically:

- Running the build process will **check for broken links**, please read the output and address any
  broken links that you are able to do.
- [This GitHub Action](https://github.com/airbytehq/airbyte/blob/master/.github/workflows/doc-link-check.yml)
  checks all links on Airbyte production docs site, and tells us if any of them are broken.

> [!NOTE] Docusaurus links checker only checks _relative_ links, and assumes that absolute links are
> fine. For that reason, if you're linking to another Airbyte documentation page, make it a relative
> link. I.e. `[link](/connector-development/overview.md)` instead of

> > `[link](https://docs.airbyte.com/connector-development/)`. That way, if your link breaks in the
> > future, it will be caught with `yarn build`.

Reviewer: ```suggestion
> `[link](https://docs.airbyte.com/connector-development/)`. That way, if your link breaks in the
> future due to a navigation restructure, it will be caught with `yarn build`.
```
probably the most likely reason

> ## Docusaurus Plusings We Use

Reviewer: ```suggestion
## Docusaurus Plugins We Use
```

Author: 🤦🏼

### Plugin Client Redirects

A silly name, but a useful plugin that adds redirect functionality to docusaurus
[Official documentation here](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-client-redirects)

> If you're proposing to move an existing documentation file or change it's name, please setup a

Reviewer: ```suggestion
If you're proposing to move an existing documentation file or change its name, please setup a
```
